v2/validator.go

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package v2
+
+import (
+	"github.com/go-playground/locales/en"
+	ut "github.com/go-playground/universal-translator"
+
+	"github.com/go-playground/validator/v10"
+)
+
+var (
+	val   *validator.Validate
+	trans ut.Translator
+)
+
+func NewValidator() *validator.Validate {
+	translator := en.New()
+	uni := ut.New(translator, translator)
+	trans, _ = uni.GetTranslator("en")
+	val = validator.New()
+	_ = val.RegisterTranslation("required", trans, func(ut ut.Translator) error {
+		return ut.Add("required", "{0} cannot be blank", true) // see universal-translator for details
+	}, func(ut ut.Translator, fe validator.FieldError) string {
+		t, _ := ut.T("required", fe.Field())
+		return t
+	})
+	return val
+}
+
+func Validate(a interface{}) error {
+	err := val.Struct(a)
```

Comment:
Is this doing validation? if so, what is it validating?

Reply:
@lenny-intel we are using the new validation library.  the validation rules define in the Struct comment
For example,
```
type Request struct {
	CorrelationID string `json:"correlationId" validate:"required,uuid"`
	RequestID     string `json:"requestId" validate:"required,uuid"`
}
```

Reply:
https://godoc.org/github.com/go-playground/validator